lib/pavilion/output.py

```diff
@@ -408,9 +451,11 @@ def __format__(self, format_spec):
 }
 
 
-def draw_table(outfile, fields, rows,
-               field_info=None, border=False, pad=True,
-               border_chars=None, header=True, title=None, table_width=None):
+def draw_table(outfile: TextIO, fields: List[str], rows: Dict[str, Any],
+               field_info: Optional[Dict[str, Dict]] = None, border: bool = False,
+               pad: bool = True, border_chars: Optional[Dict[str, str]] = None,
+               header: bool = True, title: str = None, table_width: Optional[int] = None,
+               num_digits: Optional[int] = None) -> None:
```

Comment:
Change 'num_digits' to 'default_format', which should take a function.  Then just pass your function.

Reply:
You can then just set `default_format` to be the default when the formatter is pulled out of the field_info dict.